### ISSUE TITLE
Any plans to compile this? I really need this for my Android TV

### ISSUE THREAD

**Issue by anon238** [dhk2/BitchuteTV#1]

**Comment by dhk2** [comment]
I had paused development until they released the announced APK, They still haven't released it and it's way behind schedule.
You should be able to install from 
https://github.com/dhk2/BitchuteTV/blob/master/app/release/app-release.apk

It's wonky the first few runs as it updates the local DB. Definitely not done, but it seems to still be working on my virtual devices

**Comment by anon238** [comment]
Dope! I just checked with my MiBox 4 and I had only to restart the app once until I saw all categories. App is still wonky indeed, but it’s working for the basics. Maybe some finishing lines and login integration. I have just checked quickly, but I can help you test and help thinking with you for new ideas. Overall thanks for this. Bitslide currently not supporting native Android TV.

**Comment by dhk2** [comment]
alright, updated for how bitchute changed the hashtags and trending behavior
made it a little more responsive to updates as well
Too bad they had to delay the API, because that'd make the whole process much easier.
I'll work on finishing the subscription import feature and static category/hashtag rows for basic functionality
looking into updating android tv with information on the channels.

**Comment by dhk2** [comment]
seem to have gotten the the import subscriptions feature working
it's a pain on my virtual tv, because the keyboard covers the password prompt which means blind typing, and the bitchute logon page doesn't work if you just hit enter on the password. you have to hide the keyboard and then click the button

**Comment by anon238** [comment]
> seem to have gotten the the import subscriptions feature working
> it's a pain on my virtual tv, because the keyboard covers the password prompt which means blind typing, and the bitchute logon page doesn't work if you just hit enter on the password. you have to hide the keyboard and then click the button

I can test it with my MiBox?

**Comment by dhk2** [comment]
Yeah, just install the latest release build and scroll down to the bottom

**Comment by cw-kid** [comment]
Hi

This looks like a good app. Is it the only one for Android TV / FireTV? Its the first one I found.

I've got logged in and can see my subscription channels OK.

Some channels only show a few videos and not all that are on the website for some reason. 

Also the video thumbnails with the title text, can those titles be made scrollable / ticker tape, so you can read the full video title text? 

Keep up the good work ! 

Thanks

**Comment by anon238** [comment]
> Hi
> 
> This looks like a good app. Is it the only one for Android TV / FireTV? Its the first one I found.
> 
> I've got logged in and can see my subscription channels OK.
> 
> Some channels only show a few videos and not all that are on the website for some reason.
> 
> Also the video thumbnails with the title text, can those titles be made scrollable / ticker tape, so you can read the full video title text?
> 
> Keep up the good work !
> 
> Thanks

It’s not stable enough to use the app tbh. You can better use bitslide until @dhk2 fix the bugs.

**Comment by cw-kid** [comment]
> > Hi
> > This looks like a good app. Is it the only one for Android TV / FireTV? Its the first one I found.
> > I've got logged in and can see my subscription channels OK.
> > Some channels only show a few videos and not all that are on the website for some reason.
> > Also the video thumbnails with the title text, can those titles be made scrollable / ticker tape, so you can read the full video title text?
> > Keep up the good work !
> > Thanks
> 
> It’s not stable enough to use the app tbh. You can better use bitslide until @dhk2 fix the bugs.

I did actually try BitSlide first. I side loaded it on to my Amazon Fire stick. 

However its not usable, you can't navigate and operate the app properly with a remote control as it's designed for touch screens not TV screens. 

Even with a mouse pointer I struggled. 

Main issue was not being able to move up and down on the thumbnail list of videos after you have played your first video, it seemed to stop working.

**Comment by anon238** [comment]
Yeah Bitslide is not working fine on Android TV as well.

**Comment by dhk2** [comment]
> Hi
> This looks like a good app. Is it the only one for Android TV / FireTV? Its the first one I found.

I haven't seen any others yet. I'm sure when the Bitslide guys release their version it will be much better :)

> Some channels only show a few videos and not all that are on the website for some reason.

Yeah, Fixing that would be a lot easier if Bitchute released the API last year like they originally intended. I'll look into further optimizing the channel synching

> Also the video thumbnails with the title text, can those titles be made scrollable / ticker tape, so you can read the full video title text?

Done, I just updated the release APK

**Comment by cw-kid** [comment]
> I'm sure when the Bitslide guys release their version it will be much better

So have those guys said they will have a version for TV screens ? 

> Done, I just updated the release APK

That's great thank you, much better now you can see and read the entire title of the videos. 

Another thing you maybe able to improve and this seems to be an intermittent issue as some times it doesn't happen and sometimes it does. 

When you click in to a video and you are on its details page where you can see the Watch and Channel buttons and the videos synopsis text etc, if you press back on the remote to return to the list of videos, sometimes it doesn't remain on the video tile you just clicked in to, instead it jumps back fully to the left to the very first video in the list. 

So then you have to scroll right again through all the videos you just looked at, to find where in the video list you were up to with your browsing, which is kind of annoying. 

However other times when you press back to exit from the videos details page, it does remain on the correct video tile in the list and does not jump back to the very first video. 

Thank you.

**Comment by anon238** [comment]
> > I'm sure when the Bitslide guys release their version it will be much better
> 
> So have those guys said they will have a version for TV screens ?
> 
> > Done, I just updated the release APK
> 
> That's great thank you, much better now you can see and read the entire title of the videos.
> 
> Another thing you maybe able to improve and this seems to be an intermittent issue as some times it doesn't happen and sometimes it does.
> 
> When you click in to a video and you are on its details page where you can see the Watch and Channel buttons and the videos synopsis text etc, if you press back on the remote to return to the list of videos, sometimes it doesn't remain on the video tile you just clicked in to, instead it jumps back fully to the left to the very first video in the list.
> 
> So then you have to scroll right again through all the videos you just looked at, to find where in the video list you were up to with your browsing, which is kind of annoying.
> 
> However other times when you press back to exit from the videos details page, it does remain on the correct video tile in the list and does not jump back to the very first video.
> 
> Thank you.

Those annoying bugs exists only because Bitchute doesn’t release their API making it more time to debug everything.

**Comment by cw-kid** [comment]
OK I've been using the app all evening now to watch videos.

Very informative and somewhat scary what I have learnt this evening. 

Anyway I am using the "Subscribed Videos" menu item.

After each video I played and there was a lot of them, when pressing back on the remote to return to the Subscribed videos list, each time I was then taken back to the very first video at the far left. 

So this meant having to scroll and scroll and scroll right again to get back to where I was previously in my browsing and viewing. 

If there is anyway you can stop this from happening and for it to just remain on the video tile of the video I just watched / backed out of, this would greatly improve the usability of the app. 

Many thanks for your efforts on this important project.

**Comment by anon238** [comment]
The biggest issue in the app is not showing all video's and not sorted by
newest. Most video's even in subscribed are shuffled from wayback. I think
the developer is very aware that his app is still wonky.

Op za 9 jan. 2021 om 21:56 schreef
reply@reply.github.com via
SimpleLogin <ra+nwbqgpeowxsrflrafseeddvixzffhe@simplelogin.co>:

**Comment by dhk2** [comment]
I think i've nailed the losing place issue in the video lists. It had to do if any data was updated in the background which is why it didn't always happen. Latest build uploaded should fix it.

The channels not populating fully issue is one of the things I was waiting on the API for, if i get a long stretch of freetime I'll try some work arounds.

The sorting problem has to do with the non-standard RSS feeds bitchute implemented. looking into some work arounds

**Comment by cw-kid** [comment]
> I think i've nailed the losing place issue in the video lists

Excellent work thank you ! This makes the app usable for me. 

A quick initial test (Using the Subscribed Videos menu item) and it does now appear to not be always jumping back to the very first video at the far left, but remaining positioned on the current videos tile. So looks good so far. 

EDIT: Actually it does sometimes jump back to the first video but then it auto scrolls right itself to get back to the last video you just watched or were looking at the details page for. Looks a bit weird when it happens, but I can live with it, at least it now automatically takes me back to the place in the list I was up to. So much more usable. 

I have noticed on some videos when you enter into their details page, the Watch and Channel buttons are missing and you then cannot play that video. 

EDIT: Seems to be videos from a channel called "Just the News" that do this and have the missing buttons. I will check if videos from other channels have this problem also? 

EDIT2 It happening with videos from other channels also. I think its a timing issue, If you wait long enough on the video details page the missing Watch and Channel buttons then do appear. Or you can exit the video details page and go back in to it again and that sometimes makes the missing buttons appear. Most of the time it works OK and the two buttons are present and visible right away. 

"Just the News" channel the buttons never appear on those videos, so must be a problem with that particular channel. 

This happens on some videos, but most do have the buttons OK and play OK. 

Thanks.

**Comment by anon238** [comment]
I’m gonna test this in a bit! The sorting cannot be fixed?

**Comment by cw-kid** [comment]
If I had one feature request at the moment, it would be to be able to add videos to a favourites list, so I can easily find those videos again at a later date for additional viewing. 

Also if possible a watched tick icon if possible on the videos thumbnail if you have watched that video already. 

I see you are still working on the Search feature also, as per your popup message in the app.

**Comment by cw-kid** [comment]
There is a slight bug with the video titles text if they contain Quotes " " 

They display `&quot;` rather than displaying the quotes characters I think.

**Comment by cw-kid** [comment]
Sometimes the placement / auto scrolling fix jumps you further forward in to the current video list, rather than remaining on the current video. 

Try this as a test,, go to Categories from the main menu on the left, then select the News & Politics category. 

Side Note - Once you have clicked OK on a category tile it positions you back to the first category tile "Anime & Animation" it doesn't actually take you to the category you clicked OK on. So instead I press up twice on the remote control and then I can see the "News & Politics" video list. 

Now scroll right a bit and click in to one of the videos. Wait on the video details page for a while without doing anything. Then click back or start playing a video and then stopping it and then click back.

Some times it will remain on that videos tile OK as expected, other times however its auto scrolling / jumping me much further forward in to the video list and positioning me there instead of at the current video I just viewed / played.

You may have to try it several times on various videos to see it happening. 

Again this is intermittent, some times it works and remains on the video tile you were just looking at, but some times it scrolls you forward way past that video tile, 

I don't appear to have this issue when using the "Subscribed Videos" menu item instead of using Categories. 

With Subscribed Videos menu video list, I haven't seen it jump forward to the wrong position and it works as expected.

Seems the auto scrolling / positioning doesnt work in all areas of the app? 

Go to "Subscribed Channels" menu item, then go into a channel. Move right a bit then click OK to enter a videos details page.

Then click back, some times it will remain on that videos tile and other times you are taken back to the very first video on the far left.

**Comment by cw-kid** [comment]
Try unsubscribimg from a channel via the Bitchute website.

That channel still remains in the app under Subscribed channels, even after waiting for some time and also rebooting the FireTV stick.

Thanks

**Comment by cw-kid** [comment]
App not playing any videos on a second FireTV 4K stick at a different location.

I managed to install the app OK and login OK to my account to do the import. 

However I am unable to play any videos at all.

I can't even view the video details pages properly. I click in to a video, it then briefly shows the details page where the Watch and Channel buttons are seen and then it bombs out back to the apps main home screen. 

On the rare occasion I was able to press the watch button, I just get a spinning circle and nothing starts to play. 

The WiFi at this current location is working and content in Kodi is playing / Streaming OK.

I thought maybe their ISP might be the problem, but I've tried using the Bitchute app with an active VPN connection on the FireTV stick and it still behaved very strange with the videos not playing or the videos details page just vanishing shortly after you enter in to it. 

I didn't see any of these issues on my own FireTV 4K stick at my home location.

So not sure what's going on.

**Comment by dhk2** [comment]
Alright, that seems to have sorted out the sorting problem.
The problem where it would zoom to another place in the list looks like it was caused by a bug that was duplicating videos in the database. Fixed that and added a button to clean out dupes in case i didn't fix it as well as i hope

Yeah, with the way bitchute doesn't supply the video link from the videos list It has to open the video page and update the database in the background, if there are a lot of background tasks taking place it can take a bit, and if there's a network issue or the DDOS protection starts blocking then it can take much longer.

The subscriptions aren't live linked, that's another thing that could be done with a lot of work but would be wasted effort once the API is released

That does sound strange, I'll hook up my old firestick and see if it replicates the problem.

Yeah, there's some weirdness with the text encoding and jsoup that cause quotes and ampersands to get screwy

thanks for all the feedback

**Comment by cw-kid** [comment]
Have you guys had any issues with video playback just randomly stopping?

This happens sometimes, playback stops. The video image remains on screen however. Pressing play on the remote or selecting the play / pause button in the GUI doesn't resume playback.

I have to press back to exit to the video details page and then press the watch button again. 

The video then starts again from the beginning and I have to skip the video forward to get back to where I was previously.

Thanks

**Comment by anon238** [comment]
All the time.

**Comment by cw-kid** [comment]
> The problem where it would zoom to another place in the list looks like it was caused by a bug that was duplicating videos in the database. Fixed that and added a button to clean out dupes in case i didn't fix it as well as i hope

Yes I did notice some videos where being duplicated in the lists. 

It seems to be behaving better now after your last update and not jumping forward in the video list that I have seen. 

Thanks

**Comment by cw-kid** [comment]
Just watched a 30 minute video and playback stopped 5 times. 😢

Had to back out to the video details page and press the Watch button again and then skip forward to last viewing position.

**Comment by dhk2** [comment]
Thanks for the heads up on the playback issue. I have upgraded the video player to exoplayer instead of the built in video player.
Hopefully this will enable better stability. Unfortunately with bitchute, I frequently waste time hunting down bugs that are just the site being flakey :)

**Comment by cw-kid** [comment]
Thanks for the update it appears to be playing much better now. As yet I've not seen playback just stop in the middle of a video etc.

**Comment by anon238** [comment]
> Thanks for the update it appears to be playing much better now. As yet I've not seen playback just stop in the middle of a video etc.

I still have this issue, did you updated the `apk-release.apk`? I have updated it today, but it seems its the same package or the bug isn’t removed from it. Also is it possible to create an update button internally?
